srcpkgs/recutils/patches/copyfile.patch

```diff
@@ -0,0 +1,37 @@
+--- a/utils/recutl.c
++++ b/utils/recutl.c
+@@ -58,6 +58,33 @@
+ #include <recutl.h>
+ #include "read-file.h"
+ 
++#include <sys/wait.h>
++#include <stdio.h>
++#include <stdlib.h>
++#include <unistd.h>
++
++static int qcopy_file_preserving(const char *src, const char *dst)
++{
++        int stat_loc;
++        pid_t pid = fork();
++        if (pid < 0) {
++                perror("fork");
++                return -1;
++        }
++        if (pid == 0) {
++                execl("/bin/cp", "--", src, dst, (void *)0);
++                exit(1);
```

Comment:
```suggestion
+                _exit(1);
```

Calling `exit` in a forked child can lead to all sorts of weirdness ;)

Reply:
Ouch, there are a couple of calls to `atexit`, good call!